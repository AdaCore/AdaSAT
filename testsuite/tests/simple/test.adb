with Ada.Text_IO; use Ada.Text_IO;

with AdaSAT.Formulas;
with AdaSAT.Helpers;

procedure Test is
   use AdaSAT;
   use AdaSAT.Formulas;

   F : Formula;
   M : Model := (1 .. 7 => Unset);
begin
   F.Append (new Literal_Array'(-1, +2));
   F.Append (new Literal_Array'(-3, +4));
   F.Append (new Literal_Array'(-6, -5, -2));
   F.Append (new Literal_Array'(-5, +6));
   F.Append (new Literal_Array'(+5, +7));
   F.Append (new Literal_Array'(-1, +5, -7));
   if Helpers.DPLL_Solve (F, M) then
      Put_Line ("Solved");
   else
      Put_Line ("Failed solving");
   end if;
   for E of M loop
      Put (E'Image);
      Put (" ");
   end loop;
   New_Line;
end Test;
