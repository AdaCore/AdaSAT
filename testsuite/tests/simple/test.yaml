project_file: prj.gpr
executable: simple
