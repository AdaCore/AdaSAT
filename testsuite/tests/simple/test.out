Solved
FALSE TRUE TRUE TRUE FALSE TRUE TRUE 
